hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/AmqpMessageHandlerService.java

```diff
@@ -215,14 +221,38 @@ private void registerTarget(final Message message, final String virtualHost) {
 
         try {
             final URI amqpUri = IpUtil.createAmqpUri(virtualHost, replyTo);
-            final Target target = controllerManagement.findOrRegisterTargetIfItDoesNotExist(thingId, amqpUri);
+            final Target target;
+            if (isOptionalMessageBodyEmpty(message)) {
+                target = controllerManagement.findOrRegisterTargetIfItDoesNotExist(thingId, amqpUri);
+            } else {
+                checkContentTypeJson(message);
+
+                    target = controllerManagement.findOrRegisterTargetIfItDoesNotExist(thingId, amqpUri, tryConvertThingCreatedMessage(message).getName());
+
+            }
             LOG.debug("Target {} reported online state.", thingId);
             sendUpdateCommandToTarget(target);
-        } catch (EntityAlreadyExistsException e) {
-            throw new AmqpRejectAndDontRequeueException("Target already registered, message will be ignored!", e);
+        } catch (final EntityAlreadyExistsException e) {
+            throw new AmqpRejectAndDontRequeueException(
+                    "Tried to register previously registered target, message will be ignored!", e);
         }
     }
 
+    private DmfCreateThing tryConvertThingCreatedMessage (Message message){
```

Comment:
you do not need this method, because MessageConversionException is already checked in `DelayedRequeueExceptionStrategy` and will be rejected either way

Reply:
I removed this method `tryConvertThingCreatedMessage()` and convert the message directly when calling the method `findOrRegisterTargetIfItDoesNotExist()`.